app/controllers/users/transfers_controller.rb

```diff
@@ -19,7 +19,10 @@ def update
     end
 
     def destroy
-      transfer = DossierTransfer.find_by!(id: params[:id], email: current_user.email)
+      transfer = DossierTransfer.find(params[:id])
+
+      authorized_email = (transfer.email == current_user.email || transfer.dossiers.where(dossiers: { user: current_user }).present?)
```

Comment:
je me demandais si je pouvais simplifier `transfer.dossiers.where(dossiers: { user: current_user }).present?`par `transfer.dossiers.first.user.email`

Reply:
J'ai plutôt tendance à préférer la clarté et l'intention explicite sur une action de destruction.
A priori on pourrait faire comme tu le décris oui avec un .first, j'ai vérifié tous les dossiers sont tjs en lien avec le même user, donc le premier suffit (cependant je trouve ça un peu étrange sur une relation has_many , j'ai d'ailleurs du  vérifier en base que c'est correct…).


Ça pourrait aussi s'écrire/ se simplifier avec `exists?` (qui évite d'ailleurs de charger tous les dossiers) : 

```ruby
authorized = transfer.email == current_user.email || transfer.dossiers.exists?(dossiers: { user: current_user })
render :unauthorized and return if !authorized
```

En revanche le return tel quel n'est pas le bienvenu tel quel dans un controller, car rails ne renverrait pas de réponse correcte ici. Il faut spécifier un render avec le bon status, et ensuite faire le return comme ceci avec un `and` plutôt qu'un `&&` (pour des soucis de précédente à cause du if qui vient après). Ou bien un flash avec redirection pour éviter une page blanche, mais c'est assez secondaire…

Je me demande s'il ne faudrait pas cabler un test aussi pour tester cette partie là qui est mine de rien pas trivial